### DISCUSSION TITLE
Resolving a Promise in Go

### DISCUSSION THREAD

**Discussion by zllovesuki** [dop251/goja#489] [Q&A]
Is there a way to handle resolution of a Promise created by JavaScript from Go?

With v8, there's `v8::Isolate::PerformMicrotaskCheckpoint()` to run in a loop to wait for resolution. Currently the way I'm using a helper function that takes callbacks in JavaScript:

```typescript
const __runtimeResolverResult = <T>(
  arg: T | Promise<T>, // argument as native object
  resolve: (result: T) => void, // callback to Go when the Promise resolves
  reject: (e: unknown) => void // callback to Go when the Promise rejects
) => {
  Promise.resolve(arg).then(resolve).catch(reject);
};
```

(Note that I do have `eventloop` from `https://github.com/dop251/goja_nodejs` as well).

**Comment by dop251** [comment]
I'm not sure I follow... You can set any Go function as a Promise reaction. There is no direct API for this, but you can use the generic one, i.e. promise.Get("then"), AssertFunction(), call that function with your handler as the argument.
